server/client.go

```diff
@@ -4254,23 +4336,54 @@ func (c *client) processServiceImport(si *serviceImport, acc *Account, msg []byt
 	var lrts [routeTargetInit]routeTarget
 	c.in.rts = lrts[:0]
 
+	var skipProcessing bool
+	// If message tracing enabled, add the service import trace.
+	if mt != nil {
+		mt.addServiceImportEvent(siAcc.GetName(), string(pacopy.subject), to)
+		// If we are not sharing and doing trace only, we stop at this level.
+		if !share {
```

Comment:
As noted I messed this up, let's switch to allow_trace for service export and stream import.